src/ops/async/wait-race.ts

```diff
@@ -66,60 +66,57 @@ export function waitRaceAsync<T>(
         [$A](): AsyncIterator<T> {
             const source = iterable[$A]();
             let finished = false;
-            // resolvers for currently active tasks,  which are racing to remove and call the first one
-            const resolvers:
-                | null
-                | ((res: IteratorResult<T> | Promise<never>) => void)[] = [];
-            // cache of promises to be resolved or to be returned by `.next()` to the destination
+            // resolvers for currently active tasks, that are racing to remove and call the first one:
+            const resolvers: ((
+                res: IteratorResult<T> | Promise<never>
+            ) => void)[] = [];
+            // cache of promises to be resolved or to be returned by `.next()` to the destination:
             const promises: Promise<IteratorResult<T>>[] = [];
-
             function kickOffNext(): void {
                 promises.push(
-                    new Promise((resolve, reject) => {
-                        resolvers?.push(resolve);
-
+                    new Promise((resolve) => {
+                        resolvers.push(resolve);
                         // `new Promise` executor handles synchronous exceptions
-                        source.next().then(
-                            (a) => {
+                        source
+                            .next()
+                            .then((a) => {
                                 if (a.done) {
                                     finished = true;
-                                    resolvers?.pop()?.(a);
+                                    resolvers.pop()!(a);
                                 } else if (isPromiseLike(a.value)) {
                                     const promise = Promise.resolve(a.value);
-                                    promise.then(
-                                        (value: any) => {
-                                            resolvers?.shift()?.({
+                                    promise
+                                        .then((value: any) => {
+                                            resolvers.shift()!({
                                                 done: false,
                                                 value,
                                             });
                                             kickOffMore();
-                                        },
-                                        (_) => {
-                                            resolvers?.shift()?.(
-                                                promise as any
+                                        })
+                                        .catch(() => {
```

Comment:
The [two-argument version of `then`](https://stackoverflow.com/q/24662289) was used on purpose here. Not only is it more efficient since it creates only a single extra promise, it also is important for microtask timing so that rejections are forwarded as fast as fulfillments are. Using `.catch()`, they take an extra tick.
(Same below on the `source.next()` promise)